lib/components/Editor/CustomExtensions/Mention/utils.js

```diff
@@ -1,11 +1,23 @@
+import Fuse from "fuse.js";
+import { pluck } from "ramda";
+
 export const formatMentions = (users = []) =>
   users.sort((user1, user2) => user1.name.localeCompare(user2.name));
 
 export const createMentionSuggestions = (users = []) => {
   const mentions = formatMentions(users);
 
-  return ({ query }) =>
-    mentions.filter(({ name }) =>
-      name.toLocaleLowerCase().includes(query.toLocaleLowerCase())
-    );
+  return ({ query }) => {
+    if (!query) {
+      return mentions;
+    }
+
+    const options = {
+      keys: ["name"],
+    };
+    const fuse = new Fuse(mentions, options);
+    const results = fuse.search(query);
+
+    return pluck("item", results);
+  };
```

Comment:
I used the fuzzy mentions feature for a seed data of 5000 users.

1. The behaviour for low threshold is what is expected. Add a `threshold` of `0.25`.
2. The current logic impacts the performance greatly. Slice the resulting array to 10 items or below.

@gaagul _a

Reply:
Fuse is quite slow and produces poor matches. you might consider something else:

https://github.com/leeoniya/uFuzzy#a-biased-appraisal-of-similar-work

Reply:
@leeoniya thank you for the suggestion. We'll definitely check out uFuzzy. For now we'll go forward with Fuse.js. We'll come back after creating a POC with uFuzzy and verifying the performance.